6_nodejs/week3/app.js

```diff
@@ -0,0 +1,62 @@
+
+
+
+
+const knex = require("knex")({
+    client: "mysql2",
+    connection: {
+      host: process.env.DB_HOST || "127.0.0.1",
+      port: process.env.DB_PORT || 3306,
+      user: process.env.DB_USER || "root",
+      password: process.env.DB_PASSWORD || "password1",
+      database: process.env.DB_NAME || "hyf_node_week3_warmup",
+      multipleStatements: true,
+    },
+  });
+  
+  const express = require("express");
+  const app = express();
+  const port = process.env.PORT || 3000;
+  
+  app.use(express.json());
+  
+  const apiRouter = express.Router();
+  app.use("/api", apiRouter);
+  
+  const contactsAPIRouter = express.Router();
+  apiRouter.use("/contacts", contactsAPIRouter);
+  
+  
+  // URL with injections for test http://localhost:3000/api/contacts?sort=last_name;%20DROP%20TABLE%20contacts;%20DESC
+  
+  contactsAPIRouter.get("/", async (req, res) => {
+    let query = knex.select("*").from("contacts");
+  
+    if ("sort" in req.query) {
+        const orderBy = req.query.sort.toString();
+        if (orderBy.length > 0) {
+            const [column, order] = orderBy.split(' ');
+            if (column && order && (order.toUpperCase() === 'ASC' || order.toUpperCase() === 'DESC')) {
+                query = query.orderBy(column, order);
+            } else {
+                res.status(400).json({ error: "Invalid sort parameter" });
+                return;
+            }
+        }
+    }
+  
+    const sql = query.toSQL().toNative();
+    console.log("SQL", sql);
+  
+    try {
+        const data = await query;

```

Comment:
I think the correct way to call query should be `query()`, i.e. with brackets. But I'm not fully sure about this - it might be one of the new JavaScript syntax features.

In general, for backward compatibility, I recommend using empty brackets whenever you make a function call that takes no parameters :)